### DISCUSSION TITLE
Automatic task deduplication?

### DISCUSSION THREAD

**Discussion by infinityredux** [taskiq-python/taskiq#272] [Q&A] [answered]
Is there any mechanism for handling automatic task de-duplication? 

More precisely, a way of dropping/aborting/ignoring a task based on some pre-defined criteria or reference, so that only one task with a specific reference can be queued at any given time?

I can't find anything obvious in the documentation to directly support this, so would imagine this would need a middleware or likely a custom extension to a broker. If so, I assume using a label rather than a custom task id would be preferable, but there doesn't seem to be an obvious way to filter existing queued tasks by label without getting into broker specific queries.

Would appreciate any insight people can offer.

**Comment by s3rius** [comment, marked as answer]
So, most probably, you want to create a middleware that filters out messages.
Here you can see an example of how to do something similar https://github.com/taskiq-python/taskiq/issues/271.

**Reply by infinityredux** [reply]
Ah, perfect. Thanks for the help, and the quick response.
